### ISSUE TITLE
Remove Desktop UI?

### ISSUE THREAD

**Issue by RobLoach** [flathub/org.libretro.RetroArch#231]
The open Desktop Menu option requires us to use the KDE runtime. If we disable the desktop UI, then we can switch runtimes. Do people use the Desktop UI?

**Comment by ds22x** [comment]
I'd hazard a guess and say the majority of people don't use it.

**Comment by soredake** [comment]
People that use it might not know about this issue, or this bugtracker at all.

**Comment by RobLoach** [comment]
An option we do have is to flip to the other runtime, and remove the UI, and see if we get any reports of people missing it.

**Comment by soredake** [comment]
@RobLoach what's the reason for switching runtime anyway? Is it space usage?

**Comment by soredake** [comment]
I think having all app features present is important, even if small amount of users will use it, and i don't think space usage of kde runtime is a problem:
https://theevilskeleton.gitlab.io/2022/05/16/response-to-flatpak-is-not-the-future.html#size
https://blogs.gnome.org/wjjt/2021/11/24/on-flatpak-disk-usage-and-deduplication/

I've done some calculations using https://gitlab.com/TheEvilSkeleton/flatpak-dedup-checker:

Space usage with default install:

```
anya@localhost:~> ~/git/flatpak-dedup-checker/flatpak-dedup-checker --path=~/.local/share/flatpak
Directories:                /home/anya/.local/share/flatpak/{runtime,app}
Size without deduplication: 2.16 GB
Size with deduplication:    1.98 GB (91% of 2.16 GB)
Size with compression:      1.69 GB (78% of 2.16 GB; 85% of 1.98 GB)
anya@localhost:~> 
```

Space usage after removing kde runtime's (`org.kde.Platform` and `org.kde.Platform.Locale`) and installing `org.freedesktop.Platform` and `org.freedesktop.Platform.Locale`:
```
anya@localhost:~> ~/git/flatpak-dedup-checker/flatpak-dedup-checker --path=~/.local/share/flatpak
Directories:                /home/anya/.local/share/flatpak/{runtime,app}
Size without deduplication: 3.20 GB
Size with deduplication:    2.33 GB (72% of 3.20 GB)
Size with compression:      2.01 GB (62% of 3.20 GB; 86% of 2.33 GB)
```

Space usage with retroarch and `org.kde.Platform org.kde.Platform.Locale org.freedesktop.Platform org.freedesktop.Platform.Locale`:
```
anya@localhost:~> ~/git/flatpak-dedup-checker/flatpak-dedup-checker --path=~/.local/share/flatpak
Directories:                /home/anya/.local/share/flatpak/{runtime,app}
Size without deduplication: 5.46 GB
Size with deduplication:    2.72 GB (49% of 5.46 GB)
Size with compression:      2.27 GB (41% of 5.46 GB; 83% of 2.72 GB)

```

Space usage with retroarch and `org.gnome.Platform org.gnome.Platform.Locale org.freedesktop.Platform org.freedesktop.Platform.Locale`:
```
anya@localhost:~> ~/git/flatpak-dedup-checker/flatpak-dedup-checker --path=~/.local/share/flatpak
Directories:                /home/anya/.local/share/flatpak/{runtime,app}
Size without deduplication: 5.33 GB
Size with deduplication:    2.60 GB (48% of 5.33 GB)
Size with compression:      2.24 GB (42% of 5.33 GB; 86% of 2.60 GB)
```

Space usage with retroarch and `org.gnome.Platform org.gnome.Platform.Locale org.kde.Platform org.kde.Platform.Locale org.freedesktop.Platform org.freedesktop.Platform.Locale`:
```
anya@localhost:~> ~/git/flatpak-dedup-checker/flatpak-dedup-checker --path=~/.local/share/flatpak
Directories:                /home/anya/.local/share/flatpak/{runtime,app}
Size without deduplication: 7.59 GB
Size with deduplication:    2.98 GB (39% of 7.59 GB)
Size with compression:      2.50 GB (32% of 7.59 GB; 83% of 2.98 GB)
```

I retroarch is the only thing that is installed from flathub space usage will be bigger, if use have more apps, kde or gnome, deduplication will be even more effective.

So, the question is, is space usage of kde runtime really a problem, considering deduplication is used by default, and that with btrfs compression space usage is even lower?

**Comment by ssokolow** [comment]
> People that use it might not know about this issue, or this bugtracker at all.

I was completely unaware of the Desktop UI.

In fact, my belief that it didn't exist was one of the reasons I haven't switched my use of RetroArch to Flatpak distribution when practically everything else *is* on Flatpak.

I find the default UI so frustrating and awkward that I hacked together my own ROM-launching solution rather than using the default UI and my workaround for [libretro/RetroArch#746](https://github.com/libretro/RetroArch/issues/746) relies on the system install to be able to see and construct `-L` arguments using explicit libretro core paths.

(In fact, I'm so dissatisfied with the default UI and how awkward I find its navigation layout that I prefer to use anything but libretro where viable... usually Mednafen and, for the Batocera-based TV box I built my brother, I warned him to "watch that step. It's a doozy" when he needs to go beyond the configuration and navigation stuff that EmulationStation provides its own replacement for.)
